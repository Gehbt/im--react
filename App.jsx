const App = <div id="app">Hi trans by vite</div>;
console.log('App :>> ', App);
export default App;
