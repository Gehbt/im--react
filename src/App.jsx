// import { Hypotenuse, Hypotenuse2 } from "../examples/custom_jsx.jsx";

const App = <div id="app">Hi trans by vite</div>;
// console.log("jsx Hypotenuse", <Hypotenuse a={3} b={4} />);
// console.log("jsx Hypotenuse2", <Hypotenuse2 values={[6, 8]} />);

// console.log("App :>> ", App);
export default App;
