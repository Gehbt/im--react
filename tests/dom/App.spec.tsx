// @vitest-environment jsdom
import App from "./fixtures/App";
import IMReactDOM from "~core/im-react-dom";

suite("render im-element in jsdom", () => {
  beforeAll(() => {
    vi.useFakeTimers({ toFake: ["requestIdleCallback"] });
    window.requestIdleCallback = vi.fn((cb: IdleRequestCallback) => {
      cb({
        timeRemaining: () => 12,
        didTimeout: false,
      });
      return 12;
    });
  });

  beforeEach(() => {
    vi.runAllTimers();
  });

  test("App", () => {
    // #region <----- dom ----
    const root = (function () {
      const _div = document.createElement("div");
      _div.id = "root";
      return _div;
    })();

    vi.runAllTimers();
    IMReactDOM.createRoot(root).render(<App />);
    // #endregion ----- dom ---->

    expect(root).toMatchSnapshot();
  });
});
