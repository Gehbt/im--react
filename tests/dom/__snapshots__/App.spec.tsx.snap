// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`render im-element in jsdom > App 1`] = `
<div
  id="root"
>
  <!-- 123 -->
  <p>
     Hi trans by vite
  </p>
  <div
    id="next"
  >
    <p>
       Hi trans by vite next?
    </p>
  </div>
  <span>
    Fragments
  </span>
  <p>
    FragmentL1
  </p>
  <p>
    FragmentL2
  </p>
  <p>
    Fragment2
  </p>
  <hr />
  <div
    class="counter"
  >
    <button
      class="counter-button-plus"
    >
       +1 
    </button>
    <span>
       count: 
      10
       
    </span>
    <button
      class="counter-button-minus"
    >
       -1 
    </button>
  </div>
</div>
`;
